src/SLIP10Node.test.ts

```diff
@@ -0,0 +1,345 @@
+        key: `0x${Buffer.alloc(64).fill(2).toString('hex')}`,
+        curve: secp256k1,
+      });
+
+      expect(node.key).toHaveLength(88);
+      expect(node.depth).toStrictEqual(3);
+      expect(node.toJSON()).toStrictEqual({
+        depth: 3,
+        key: node.key,
+      });
+    });
+
+    it('throws if the depth is invalid', async () => {
+      const validBufferKey = Buffer.alloc(64).fill(1);
+      const inputs = [
+        -1,
+        0.1,
+        -0.1,
+        NaN,
+        Infinity,
+        '0',
+        'zero',
+        {},
+        null,
+        undefined,
+      ];
+
+      for (const input of inputs) {
+        await expect(() =>
+          SLIP10Node.create({
+            depth: input as any,
+            key: validBufferKey,
+            curve: secp256k1,
+          }),
+        ).rejects.toThrow(
+          `Invalid HD tree path depth: The depth must be a positive integer. Received: "${input}"`,
+        );
+      }
+    });
+
+    it('throws if both a derivation path and a depth are specified', async () => {
+      await expect(() =>
+        SLIP10Node.create({
+          depth: 2, // This is the correct depth, but it's still forbidden
+          derivationPath: [
+            defaultBip39NodeToken,
+            BIP44PurposeNodeToken,
+            `bip32:60'`,
+          ],
+          curve: secp256k1,
+        }),
+      ).rejects.toThrow(
+        'Invalid parameters: May not specify a depth if a derivation path is specified. The depth will be calculated from the path.',
+      );
+    });
+
+    it('throws if neither a derivation path nor a key is specified', async () => {
+      await expect(() =>
+        SLIP10Node.create({ depth: 1, curve: secp256k1 }),
+      ).rejects.toThrow(
+        'Invalid parameters: Must specify either key or derivation path.',
+      );
+    });
+
+    it('throws if the derivation path is empty', async () => {
+      await expect(() =>
+        SLIP10Node.create({ derivationPath: [] as any, curve: secp256k1 }),
+      ).rejects.toThrow(
+        'Invalid derivation path: May not specify an empty derivation path.',
+      );
+    });
+
+    it('throws if the derivation path is of depth 0 and not a single BIP-39 node', async () => {
+      await expect(() =>
+        SLIP10Node.create({
+          derivationPath: [`bip32:0'`] as any,
+          curve: secp256k1,
+        }),
+      ).rejects.toThrow(
+        'Invalid HD path segment: The segment must consist of a single BIP-39 node for depths of 0. Received: "bip32:0\'".',
+      );
+    });
+
+    it('throws if the key is neither a string nor a buffer', async () => {
+      await expect(() =>
+        SLIP10Node.create({ depth: 0, key: {} as any, curve: secp256k1 }),
+      ).rejects.toThrow(
+        'Invalid key: Must be a Buffer or string if specified. Received: "object"',
+      );
+    });
+
+    it('throws if the key is an invalid Buffer', async () => {
+      const invalidLengthBuffer = Buffer.alloc(63).fill(1);
```

Comment:
`Buffer.alloc(63, 1)` does the same iirc

Reply:
Not sure if cleaner or not though

Reply:
We do it like this everywhere, and it's just for the tests, so I think it's fine to keep it like this?

Reply:
Yeah, just food for thought